metal/classifier.py

```diff
@@ -271,10 +292,84 @@ def _set_scheduler(self, scheduler_config, optimizer):
             )
         return lr_scheduler
 
+    def _batch_evaluate(self, loader, break_ties="random", **kwargs):
+        """Evaluates the model using minibatches
+
+        Args:
+            loader: Pytorch DataLoader supplying (X,Y):
+                X: The input for the predict method
+                Y: An [n] or [n, 1] torch.Tensor or np.ndarray of target labels
+                    in {1,...,k}; can be None for cases with no ground truth
+
+        Returns:
+            Y_p: an np.ndarray of predictions
+            Y: an np.ndarray of ground truth labels
+        """
+        Y = []
+        Y_p = []
+        for batch, data in enumerate(loader):
+            X_batch, Y_batch = data
+
+            if self.config["train_config"]["use_cuda"]:
+                X_batch = X_batch.cuda()
+
+            Y_batch = self._to_numpy(Y_batch)
+
+            if Y_batch.ndim > 1:
+                Y_batch = self._break_ties(Y_batch, break_ties)
+
+            Y.append(Y_batch)
+            Y_p.append(
+                self._to_numpy(
+                    self.predict(X_batch, break_ties=break_ties, **kwargs)
+                )
+            )
+
+        Y = np.hstack(Y)
+        Y_p = np.hstack(Y_p)
+
+        return Y_p, Y
+
+    def evaluate(self, data, break_ties="random", **kwargs):
+        """Evaluates the model
+
+        Args:
+            data: either a Pytorch DataLoader or tuple supplying (X,Y):
+                X: The input for the predict method
+                Y: An [n] or [n, 1] torch.Tensor or np.ndarray of target labels
+                    in {1,...,k}
+
+        Returns:
+            Y_p: an np.ndarray of predictions
+            Y: an np.ndarray of ground truth labels
+        """
+
+        if type(data) is tuple:
+            X, Y = data
+
+            if self.config["train_config"]["use_cuda"]:
+                X = X.cuda()
+
+            Y = self._to_numpy(Y)
+
+            if Y.ndim > 1:
+                Y = self._break_ties(Y, break_ties)
+
+            Y_p = self.predict(X, break_ties=break_ties, **kwargs)
+
+        elif type(data) is DataLoader:
+            Y_p, Y = self._batch_evaluate(data, break_ties=break_ties)
+
+        else:
+            raise ValueError(
+                "Unrecognized input data structure, use tuple or DataLoader!"
```

Comment:
Not sure we need an exclamation point. :)

Reply:
Lol, sounds good.  Will update